app/assets/javascripts/application.js

```diff
@@ -13,3 +13,12 @@
 //= require rails-ujs
 //= require turbolinks
 //= require_tree .
+
```

Comment:
I see you have this code in several places. I'm assuming because it wasn't working in several of them. 

1. If you want to keep it here in this file, you need to add `//= require_self` at the top. 
1. If you want to keep it in it's own file (like you have below), add this: `//= require chat_room`

Those are both specific parts of the asset pipeline that we have't covered. I recommend #2 above, and removing it from the view code entirely.